operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/ResourceRequirementsSanitizer.java

```diff
@@ -0,0 +1,100 @@
+package io.javaoperatorsdk.operator.processing.dependent.kubernetes;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.GenericKubernetesResource;
+import io.fabric8.kubernetes.api.model.PodTemplateSpec;
+import io.fabric8.kubernetes.api.model.Quantity;
+import io.fabric8.kubernetes.api.model.ResourceRequirements;
+
+/**
+ * Sanitizes the {@link ResourceRequirements} in the containers of a pair of {@link PodTemplateSpec}
+ * instances.
+ * <p>
+ * When the sanitizer finds a mismatch in the structure of the given templates, before it gets to
+ * the nested resource limits and requests, it returns early without fixing the actual map. This is
+ * an optimization because the given templates will anyway differ at this point. This means we do
+ * not have to attempt to sanitize the resources for these use cases, since there will anyway be an
+ * update of the K8s resource.
+ * <p>
+ * The algorithm traverses the whole template structure because we need the actual and desired
+ * {@link Quantity} instances to compare their numerical amount. Using the
+ * {@link GenericKubernetesResource#get(Map, Object...)} shortcut would need to create new instances
+ * just for the sanitization check.
+ */
+class ResourceRequirementsSanitizer {
+
+  static void sanitizeResourceRequirements(final Map<String, Object> actualMap,
+      final PodTemplateSpec actualTemplate, final PodTemplateSpec desiredTemplate) {
+    if (actualTemplate == null || desiredTemplate == null) {
```

Comment:
I'm not 100% sure if all of these `null` checks are needed. I've added them for all fields that have no default value in their declaration.